### PR TITLE
refac: Remove duplicated energy_result testing code, and mirror time_series for read_and_write.

### PR DESCRIPTION
**Pull request by slb-halfspace**
# Description

Cleans up in the energy results tests, as the balance fixing added duplicated code.
Also adds a folder for proper testing of read_and_filter for energy_results which mirrors the one for time_series.
Since this code for energy is pretty simplistic, the testing isn't very exhaustive currently, but this structure is still nice to have for future extentions :)

## Pull-request quality

- [ ] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+
+@pytest.mark.parametrize(
+    "request_energy_per_es",
+    [
+        True,
+        False,
+    ],
+)
+def test_energy_read_and_filter_for_energy_results__when_requesting_energy_per_es__returns_correct_columns(
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:43]:
```suggestion
def test_read_and_filter_from_view__when_requesting_actor_is_energy_supplier__returns_results_only_for_that_energy_supplier(
```

Comment (JesperGomes) [reply source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:43]: I propose a more business oriented test (and less whitebox). You could then add data from a few differnent suppliers in the mock return value, and verify that only relevant energy supplier data is returned.

Comment (JesperGomes) [reply source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:43]: and then you could add a parametrized test that sets energy_supplier_ids to: None, <one value> and <multiple_values>. The test name coudl then be:

`test_read_and_filter_from_view__when_datahub_admin_and_no_energy_suppliers__returns_results_expected_energy_suppliers`

Comment (JesperGomes) [reply source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:43]: and a test for grid_acces_provider where you mock just the dataframe from read_energy:

`
test_read_and_filter_from_view__when_system_operator__returns_expected_results
`

and maybe this test:

`
test_read_and_filter_from_view__when_system_operator__returns_returns_result_without_energy_supplier_column
`

File: source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    testing_spec = default_data.create_energy_results_data_spec()
+
+    df_energy_per_es_v1 = energy_factory.create_energy_per_es_v1(spark, testing_spec)
+    df_energy_v1 = energy_factory.create_energy_v1(spark, testing_spec)
+
+    mock_repository = Mock()
+    mock_repository.read_energy.return_value = df_energy_v1
+    mock_repository.read_energy_per_es.return_value = df_energy_per_es_v1
+
+    standard_wholesale_fixing_scenario_args.requesting_actor_market_role = (
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:58]: I think you need to deep copy standard_wholesale_fixing_scenario_args to avoid unexpected dependencies between tests
`args = copy.deepcopy(standard_wholesale_fixing_scenario_args)`

File: source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    )
+
+    # Act
+    actual_df = read_and_filter_from_view(
+        args=standard_wholesale_fixing_scenario_args,
+        repository=energy_read_and_filter_mock_repository,
+    )
+
+    # Assert
+    for i, column in enumerate(actual_df.columns):
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:146]: what if expected has one more column than actual?

Comment (slb-halfspace) [reply source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py]: True, this was from when I was debugging which column was failing, the right one is to use:
expected_columns == actual_df.columns

File: source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        assert actual_df.count() == 0
+
+
+@pytest.mark.parametrize(
+    "energy_supplier_ids, contains_data",
+    [
+        (None, True),
+        (["1000000000000"], True),
+        (["2000000000000", "3000000000000"], True),
+        ("'ID_WITH_NO_DATA'", False),
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:119]: is this a list?

File: source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        args=standard_wholesale_fixing_scenario_args,
+        repository=energy_read_and_filter_mock_repository,
+    )
+
+    # Assert
+    assert expected_columns == actual_df.columns
+
+    assert (
+        actual_df.filter(
+            f"{DataProductColumnNames.grid_area_code} != '{grid_area_code}'"
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:198]: what if the dataframe is empty? 

Could also be something like this:

`
actual_grid_area_codes = actual.select(DataProductColumnNames.grid_area_code).distinct().collect()
assert len(actual_grid_area_codes )==1
assert actual_grid_area_codes .first()[0]== grid_area_code
`

Comment (slb-halfspace) [reply source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:198]: The dataframe being empty is checked in the step afterwards, with actual_df.count() > 0

Comment (slb-halfspace) [reply source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:198]: This specific check simple ensures we have filtered out all non-chosen grid areas.

Comment (slb-halfspace) [reply source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py]: I changed it such that they are put into variables with more descriptive names, but I still think this tests the correct things.

File: source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    actual_df = read_and_filter_from_view(
+        args=standard_wholesale_fixing_scenario_args,
+        repository=energy_read_and_filter_mock_repository,
+    )
+
+    # Assert
+    assert expected_columns == actual_df.columns
+
+    assert (
+        actual_df.filter(
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/domain/energy_results/test_energy_read_and_filter.py:101]: what if actual_df is (accidently) empty?
